metrics/web/test/dashboard/domain/entities/metrics/receive_build_day_project_metrics_updates_test.dart

```diff
@@ -0,0 +1,263 @@
+    const projectIdParam = ProjectIdParam(projectId);
+
+    final buildDayRepository = _BuildDayRepositoryMock();
+    final useCase = ReceiveBuildDayProjectMetricsUpdates(buildDayRepository);
+
+    PostExpectation<Stream<List<BuildDay>>> whenFetchBuildDaysStream() {
+      return when(buildDayRepository.projectBuildDaysInDateRangeStream(
+        projectId,
+        from: anyNamed('from'),
+      ));
+    }
+
+    BuildDay createBuildDay({
+      int successful = 0,
+      int failed = 0,
+      int unknown = 0,
+      int inProgress = 0,
+      Duration totalDuration = Duration.zero,
+      DateTime day,
+    }) {
+      return BuildDay(
+        projectId: projectId,
+        successful: successful,
+        failed: failed,
+        unknown: unknown,
+        inProgress: inProgress,
+        totalDuration: totalDuration,
+        day: day ?? DateTime.now(),
+      );
+    }
+
+    int calculateTotalNumberOfBuilds(List<BuildDay> buildDays) {
+      return buildDays.fold<int>(0, (value, buildDay) {
+        return value +
+            buildDay.successful +
+            buildDay.failed +
+            buildDay.unknown +
+            buildDay.inProgress;
+      });
+    }
+
+    Duration calculateTotalBuildsDuration(List<BuildDay> buildDays) {
+      return buildDays.fold<Duration>(Duration.zero, (value, buildDay) {
+        return value + buildDay.totalDuration;
+      });
+    }
+
+    tearDown(() {
+      reset(buildDayRepository);
+    });
+
+    test(
+      "throws an ArgumentError if the given build day repository is null",
+      () {
+        expect(
+          () => ReceiveBuildDayProjectMetricsUpdates(null),
+          throwsArgumentError,
+        );
+      },
+    );
+
+    test(
+      "successfully creates an instance with the given build day repository",
+      () {
+        expect(
+          () => ReceiveBuildDayProjectMetricsUpdates(buildDayRepository),
+          returnsNormally,
+        );
+      },
+    );
+
+    test(
+      "successfully creates an instance with the given build day repository",
+      () {
+        expect(
+          () => ReceiveBuildDayProjectMetricsUpdates(buildDayRepository),
+          returnsNormally,
+        );
+      },
+    );
+
+    test(
+      ".call() delegates to the .projectBuildDaysInDateRangeStream() with the given project id and a non-null from date",
+      () {
+        whenFetchBuildDaysStream().thenAnswer((_) => Stream.value([]));
+
+        useCase.call(projectIdParam);
+
+        verify(buildDayRepository.projectBuildDaysInDateRangeStream(
+          projectId,
+          from: argThat(isNotNull, named: 'from'),
+        )).called(once);
+      },
+    );
+
+    test(
+      ".call() returns a build day project metrics with the given project id",
+      () async {
+        whenFetchBuildDaysStream().thenAnswer((_) => Stream.value([]));
+
+        final metricsStream = useCase.call(projectIdParam);
+        final buildDayMetrics = await metricsStream.firstWhere(
```

Comment:
Should we maybe use the [emits](https://pub.dev/documentation/test_api/latest/expect/emits.html) matcher where possible that is purposed to test streams?